contracts/test/TestERC223Callback.sol

```diff
@@ -0,0 +1,51 @@
+pragma solidity 0.4.15;
+
+import '../Standards/IERC223Token.sol';
+import '../Standards/IERC223Callback.sol';
+
+
+contract TestERC223Callback is IERC223Callback {
+
+    ////////////////////////
+    // Mutable state
+    ////////////////////////
+    address private _from;
+    uint256 private _amount;
+    bytes32 private _dataKeccak;
+
+
+    ////////////////////////
+    // Constructor
+    ////////////////////////
+    function TestERC223Callback() {
+        // some "random" hash
+        _dataKeccak = sha3(address(this));
+    }
+
+    ////////////////////////
+    // Public functions
+    ////////////////////////
+    function tokenFallback(
+        address from,
+        uint256 amount,
+        bytes data
+    )
+        public
+    {
+        _from = from;
+        _amount = amount;
+        _dataKeccak = sha3(data);
```

Comment:
It would be simpler to just fire a log event. In fact, we could have a generic mock contract that has a fallback function that logs `msg.sig` and `msg.data` and then returns the next value from a preconfigured array. That should cover a lot of cases.